stan/math/prim/prob/discrete_range_lpmf.hpp

```diff
@@ -0,0 +1,96 @@
+#ifndef STAN_MATH_PRIM_PROB_DISCRETE_RANGE_LPMF_HPP
+#define STAN_MATH_PRIM_PROB_DISCRETE_RANGE_LPMF_HPP
+
+#include <stan/math/prim/meta.hpp>
+#include <stan/math/prim/err.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+#include <stan/math/prim/fun/inv.hpp>
+#include <stan/math/prim/fun/log.hpp>
+#include <stan/math/prim/fun/max_size.hpp>
+#include <stan/math/prim/fun/size_zero.hpp>
+#include <stan/math/prim/fun/value_of.hpp>
+#include <cmath>
+
+namespace stan {
+namespace math {
+
+/** \ingroup prob_dists
+ * Return the log PMF of a discrete range for the given y, lower and upper
+ * bound.
+ *
```

Comment:
This should define the density, which is p(x | lower, upper) = 1 / (upper - lower + 1)` or on the log scale, `log p(x | lower, upper) = -log(upper - lower + 1)`.